src/extension/inpage-script/index.js

```diff
@@ -61,8 +69,12 @@ if (document) {
       return window.webln
         .sendPayment(paymentRequest)
         .then((r) => {
-          console.log(r);
-          //alert(JSON.stringify(r));
+          const responseEvent = new CustomEvent("lightning:success", {
+            paymentRequest: paymentRequest,
+            response: r,
+          });
+          document.dispatchEvent(responseEvent);
+          link.dispatchEvent(responseEvent);
```

Comment:
@escapedcat @dylancom is it better to dispatch that event on the link or on the document?

Reply:
Where is this being used?Would need to understand where this is being used. Looks like currently everything is attached to `window`. What the purpose/usage here?

Reply:
if you do a `<a href="lightning:...">..</a>` this event would allow you to get notified when a payment happened. 
`link/document.addEventListener('lightning:success', ...)`

Reply:
You dispatch it on the element, link in this case. And you can catch it on the document. This way you can also listen for which element initiated it (event.target.tagName)

Reply:
does that mean it is enough to dispatch it on the link element? (and it still bubbles up to the document?)

Reply:
If you add: `bubbles: true` to the CustomEvent options, you can catch it on document.